phpBB/includes/acp/acp_language.php

```diff
@@ -19,130 +19,212 @@
 
 		$this->language_directory = basename($this->language_directory);
 		$this->language_file = basename($this->language_file);
 
-		$user->add_lang('acp/language');
-		$this->tpl_name = 'acp_language';
-		$this->page_title = 'ACP_LANGUAGE_PACKS';
+		$this->language->add_lang('acp/language');
 
 		switch ($action)
 		{
 			case 'update_details':
 
 				if (!$submit || !check_form_key($form_name))
 				{
-					trigger_error($user->lang['FORM_INVALID']. adm_back_link($this->u_action), E_USER_WARNING);
+					trigger_error($this->language->lang('FORM_INVALID'). adm_back_link($this->u_action), E_USER_WARNING);
 				}
 
 				if (!$lang_id)
 				{
-					trigger_error($user->lang['NO_LANG_ID'] . adm_back_link($this->u_action), E_USER_WARNING);
+					trigger_error($this->language->lang('NO_LANG_ID') . adm_back_link($this->u_action), E_USER_WARNING);
 				}
 
 				$sql = 'SELECT *
 					FROM ' . LANG_TABLE . "
 					WHERE lang_id = $lang_id";
-				$result = $db->sql_query($sql);
-				$row = $db->sql_fetchrow($result);
-				$db->sql_freeresult($result);
+				$result = $this->db->sql_query($sql);
+				$row = $this->db->sql_fetchrow($result);
+				$this->db->sql_freeresult($result);
 
 				$sql_ary	= array(
-					'lang_english_name'		=> $request->variable('lang_english_name', $row['lang_english_name']),
-					'lang_local_name'		=> $request->variable('lang_local_name', $row['lang_local_name'], true),
-					'lang_author'			=> $request->variable('lang_author', $row['lang_author'], true),
+					'lang_english_name'		=> $this->request->variable('lang_english_name', $row['lang_english_name']),
+					'lang_local_name'		=> $this->request->variable('lang_local_name', $row['lang_local_name'], true),
+					'lang_author'			=> $this->request->variable('lang_author', $row['lang_author'], true),
 				);
 
-				$db->sql_query('UPDATE ' . LANG_TABLE . '
-					SET ' . $db->sql_build_array('UPDATE', $sql_ary) . '
+				$this->db->sql_query('UPDATE ' . LANG_TABLE . '
+					SET ' . $this->db->sql_build_array('UPDATE', $sql_ary) . '
 					WHERE lang_id = ' . $lang_id);
 
-				$phpbb_log->add('admin', $user->data['user_id'], $user->ip, 'LOG_LANGUAGE_PACK_UPDATED', false, array($sql_ary['lang_english_name']));
+				$this->log->add('admin', $this->user->data['user_id'], $this->user->ip, 'LOG_LANGUAGE_PACK_UPDATED', false, array($sql_ary['lang_english_name']));
 
-				trigger_error($user->lang['LANGUAGE_DETAILS_UPDATED'] . adm_back_link($this->u_action));
+				trigger_error($this->language->lang('LANGUAGE_DETAILS_UPDATED') . adm_back_link($this->u_action));
 			break;
 
 			case 'details':
 
 				if (!$lang_id)
 				{
-					trigger_error($user->lang['NO_LANG_ID'] . adm_back_link($this->u_action), E_USER_WARNING);
+					trigger_error($this->language->lang('NO_LANG_ID') . adm_back_link($this->u_action), E_USER_WARNING);
 				}
 
 				$this->page_title = 'LANGUAGE_PACK_DETAILS';
 
 				$sql = 'SELECT *
 					FROM ' . LANG_TABLE . '
 					WHERE lang_id = ' . $lang_id;
```

Comment:
Could be casted to `int) $lang_id`

Reply:
Also safe without that as it's already casted to int further above.